addons/mail/static/src/components/thread_preview/thread_preview.js

```diff
@@ -25,10 +25,13 @@ export class ThreadPreview extends Component {
      * @returns {string}
      */
     image() {
+        if (!this.threadPreviewView.thread.channel) {
+            return '/mail/static/src/img/smiley/avatar.jpg';
+        }
```

Comment:
As `correspondent` was directly on Thread there was no need for it to be guarded.
I agree with you in that I don't think there are any cases where the component doesn't have a channel. Actually, ThreadPreview should rather be ChannelPreview, but I don't intend to make the change in this PR, so as long as we have to deal with thread instead of channel directly I'd rather keep some kind of guard just in case.
Be sure however that it will be addressed in a further PR 🙂